### PR TITLE
[SofaDeformable] Fix MeshSpring ForceField and Loader

### LINKED ISSUE #804: Memory leak MeshSpringForceField
The second happened by calling the MeshSpringForceField.init() method at each iteration. I simply removed the call since it wasn’t needed.

See post: https://www.sofa-framework.org/community/forum/topic/memory-leaks/

______________________________________________________

Suggested labels:

### PR DESCRIPTION
**Pull request by hugtalbot**
One memory leak was identified, use of size_t is added and fix in the init of cube
Fix #804 

______________________________________________________

This PR: 
- [ ] builds with SUCCESS for all platforms on the CI.
- [ ] does not generate new warnings.
- [ ] does not generate new unit test failures.
- [ ] does not generate new scene test failures.
- [ ] does not break API compatibility.
- [ ] is more than 1 week old (or has fast-merge label).

**Reviewers will merge only if all these checks are true.**

### CODE REVIEW

File: SofaKernel/framework/sofa/helper/io/MassSpringLoader.cpp [hunk 1]
```diff
@@ -105,6 +105,7 @@ bool MassSpringLoader::load(const char *filename)
                 msg_error("MassSpringLoader") << "fscanf function has encountered an error." ;
                 setNumMasses(0);
                 setNumSprings(0);
+                fclose(file);
```
Comment (damienmarchal) [inline SofaKernel/framework/sofa/helper/io/MassSpringLoader.cpp:108]: Isn't there RAII pattern to handle this kind of situation ?

File: SofaKernel/modules/SofaDeformable/MeshSpringForceField.inl [hunk 2]
```diff
@@ -66,8 +66,8 @@ void MeshSpringForceField<DataTypes>::addSpring(std::set<std::pair<int,int> >& s
         sset.insert(std::make_pair(m2,m1));
     }
     Real l = ((this->mstate2->read(core::ConstVecCoordId::restPosition())->getValue())[m2] - (this->mstate1->read(core::ConstVecCoordId::restPosition())->getValue())[m1]).norm();
-     this->springs.beginEdit()->push_back(typename SpringForceField<DataTypes>::Spring(m1,m2,stiffness/l, damping/l, l, noCompression.getValue()));
-      this->springs.endEdit();
+    this->springs.beginEdit()->push_back(typename SpringForceField<DataTypes>::Spring(m1,m2,stiffness/l, damping/l, l, noCompression.getValue()));
```
Comment (damienmarchal) [inline SofaKernel/modules/SofaDeformable/MeshSpringForceField.inl]: It is out of the topic of the PR but it would be nice to remove all these 'this'.

**Comment by guparan** [comment]
[ci-build][with-scene-tests]

File: SofaKernel/framework/sofa/helper/io/MassSpringLoader.cpp (hunk 1 above)
Comment (guparan) [reply SofaKernel/framework/sofa/helper/io/MassSpringLoader.cpp:108]: See https://en.cppreference.com/w/cpp/io/basic_ofstream
and https://eli.thegreenplace.net/2016/c-raii-without-exceptions/

File: SofaKernel/modules/SofaDeformable/MeshSpringForceField.inl (hunk 2 above)
Comment (guparan) [reply SofaKernel/modules/SofaDeformable/MeshSpringForceField.inl]: See https://github.com/sofa-framework/sofa/pull/782/commits/3590a55b2e7ca5988501c353bad1ecc2264e1a0c

**Comment by guparan** [comment]
Merged. Warning for changelog: Data members were renamed with `d_` prefix.

**Pull request merged**